runtime/commonMain/src/kotlinx/serialization/json/internal/TreeJsonOutput.kt

```diff
@@ -118,6 +111,28 @@ private sealed class AbstractJsonTreeOutput(
     }
 }
 
+private class JsonTreePrimitiveOutput(json: Json, nodeConsumer: (JsonElement) -> Unit) :
+    AbstractJsonTreeOutput(json, nodeConsumer) {
+    private var content: JsonElement? = null
+
+    companion object {
+        const val primitive = "primitive"
+    }
+
+    init {
+        pushTag(primitive)
+    }
+
+    override fun putElement(key: String, element: JsonElement) {
+        require(key == primitive)
```

Comment:
1) [nitpick] you can use referential comparison here :) 
2) Please add a corresponding error message 
3) It is better to use top-level const to avoid blowing the binary with unnecessary empty classes

Reply:
Well, then `JsonPrimitiveInput` should be refactored as well. I think I can share the constant between these classes